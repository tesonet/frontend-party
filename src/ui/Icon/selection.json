{"IcoMoonType":"selection","icons":[{"icon":{"paths":["M1002.724 767.9l-361.73-664.952c-58.118-97.842-199.792-97.972-257.988 0l-361.712 664.952c-59.412 99.978 12.518 226.582 128.964 226.582h723.472c116.348 0 188.406-126.502 128.994-226.582zM512 874.482c-33.076 0-60-26.924-60-60s26.924-60 60-60c33.076 0 60 26.924 60 60s-26.924 60-60 60zM572 634.482c0 33.076-26.924 60-60 60s-60-26.924-60-60v-300c0-33.076 26.924-60 60-60s60 26.924 60 60v300z"],"attrs":[{}],"isMulticolor":false,"isMulticolor2":false,"grid":0,"tags":["alert"]},"attrs":[{}],"properties":{"order":7,"id":4,"name":"alert","prevSize":32,"code":59652},"setIdx":0,"setId":4,"iconIdx":0},{"icon":{"paths":["M960.005-0.003c38.401 0 63.999 25.602 63.999 64.002v896.002c0 38.395-25.598 63.996-63.999 63.996h-576.006c-38.389 0-63.987-25.6-63.987-63.996v-192.002h127.999v128h447.995v-767.998h-447.995v128h-127.999v-192.003c0-38.4 25.598-64.002 63.987-64.002h576.006zM256 256.002v191.999h383.996v128h-383.996v191.999l-255.996-256.001 255.996-255.997z"],"attrs":[],"isMulticolor":false,"isMulticolor2":false,"grid":0,"tags":["logout"],"colorPermutations":{}},"attrs":[],"properties":{"order":2,"id":3,"name":"logout","prevSize":32,"code":59648},"setIdx":0,"setId":4,"iconIdx":1},{"icon":{"paths":["M512 896c17.664 0 32-14.336 32-32v-638.384l232.832 244.448c6.288 6.608 14.72 9.936 23.168 9.936 7.92 0 15.872-2.928 22.064-8.832 12.8-12.192 13.296-32.448 1.104-45.248l-265.904-279.168c-12.096-12.080-28.16-18.752-45.264-18.752s-33.168 6.672-45.792 19.312l-265.376 278.624c-12.192 12.8-11.696 33.056 1.104 45.248s33.072 11.696 45.248-1.104l232.816-245.472v639.392c0 17.664 14.336 32 32 32z"],"attrs":[],"isMulticolor":false,"isMulticolor2":false,"grid":0,"tags":["arrow-up"],"colorPermutations":{}},"attrs":[],"properties":{"order":3,"id":2,"name":"arrow-up","prevSize":32,"code":59649},"setIdx":0,"setId":4,"iconIdx":2},{"icon":{"paths":["M398.23 0.002c125.147 0 227.553 102.4 227.553 227.553v0 170.668h113.777c34.125 0 56.888 22.757 56.888 56.891v0 511.997c0 34.134-22.764 56.89-56.888 56.89v0h-682.669c-34.135 0-56.888-22.756-56.888-56.89v0-511.997c0-34.134 22.753-56.891 56.888-56.891v0h113.776v-170.668c0-125.153 102.405-227.553 227.564-227.553zM398.23 568.889c-31.431 0-56.899 25.47-56.899 56.888v0 170.665c0 31.422 25.468 56.89 56.899 56.89 31.409 0 56.877-25.468 56.877-56.89v0-170.665c0-31.417-25.468-56.888-56.877-56.888zM398.23 113.78c-62.585 0-113.787 51.197-113.787 113.775v0 170.668h227.563v-170.668c0-62.578-51.213-113.775-113.776-113.775z"],"attrs":[],"width":796,"isMulticolor":false,"isMulticolor2":false,"grid":0,"tags":["lock"],"colorPermutations":{}},"attrs":[],"properties":{"order":4,"id":1,"name":"lock","prevSize":32,"code":59650},"setIdx":0,"setId":4,"iconIdx":3},{"icon":{"paths":["M706.305 603.265c112.703 36.734 189.698 120.637 189.698 228.737v0 191.999h-896.002v-191.999c0-108.1 76.984-192.003 189.697-228.737 68.288 62.336 158.783 100.737 258.31 100.737 99.502 0 190.009-38.401 258.297-100.737zM448.008 0.002c141.175 0 255.997 114.812 255.997 255.997v0 64.002c0 141.18-114.822 255.997-255.997 255.997-141.198 0-256.010-114.817-256.010-255.997v0-64.002c0-141.185 114.811-255.997 256.010-255.997z"],"attrs":[],"width":896,"isMulticolor":false,"isMulticolor2":false,"grid":0,"tags":["user"],"colorPermutations":{}},"attrs":[],"properties":{"order":6,"id":0,"name":"user","prevSize":32,"code":59651},"setIdx":0,"setId":4,"iconIdx":4}],"height":1024,"metadata":{"name":"icomoon"},"preferences":{"showGlyphs":true,"showQuickUse":true,"showQuickUse2":true,"showSVGs":true,"fontPref":{"prefix":"icon-","metadata":{"fontFamily":"icomoon","majorVersion":1,"minorVersion":0},"metrics":{"emSize":1024,"baseline":6.25,"whitespace":50},"embed":false,"showSelector":false,"showMetrics":false,"showMetadata":false,"showVersion":false},"imagePref":{"prefix":"icon-","png":true,"useClassSelector":true,"color":0,"bgColor":16777215,"classSelector":".icon","height":32,"columns":16,"margin":16,"name":"icomoon"},"historySize":50,"showCodes":true,"gridSize":16}}