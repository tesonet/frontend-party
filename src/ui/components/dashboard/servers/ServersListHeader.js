import React from 'react';
import PropTypes from 'prop-types';

// Aphrodite
import { css, StyleSheet } from 'aphrodite';

export default class ServersListHeader extends React.PureComponent {
    static propTypes = {
        activeSort: PropTypes.string.isRequired,
        changeServersSort: PropTypes.func.isRequired,
    };

    render() {
        const { activeSort, changeServersSort } = this.props;
        return (
            <div className={css(styles.container)}>
                <div className={css(styles.headerTitleContainer, styles.headerTitleLeft)}>
                    <h3
                        className={css(styles.headerText, activeSort === 'name' && styles.active)}
                        onClick={changeServersSort.bind(this, 'name')}
                    >
                        SERVER
                    </h3>
                </div>
                <div className={css(styles.headerTitleContainer, styles.headerTitleRight)}>
                    <h3
                        className={css(styles.headerText, activeSort === 'distance' && styles.active)}
                        onClick={changeServersSort.bind(this, 'distance')}
                    >
                        DISTANCE
                    </h3>
                </div>
            </div>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: '58px',
        display: 'flex',
        borderTop: '1px solid #D8D8D8',
        borderBottom: '1px solid #D8D8D8',
        backgroundColor: '#F5F5F5',
        boxSizing: 'border-box',
    },
    headerTitleContainer: {
        flex: '1 1 0%',
        display: 'flex',
        alignItems: 'center',
    },
    headerTitleLeft: {
        marginLeft: '14px',
        justifyContent: 'flex-start',
    },
    headerTitleRight: {
        marginRight: '17px',
        justifyContent: 'flex-end',
    },
    headerText: {
        color: '#999999',
        fontFamily: 'Roboto, sans-serif',
        fontSize: '14px',
        fontWeight: 300,
        lineHeight: '30px',
        letterSpacing: '1.4px',
        cursor: 'pointer',
        ':active': {
            transform: 'scale(1.1)',
        }
    },
    active: {
        fontWeight: 700,
    },
});
