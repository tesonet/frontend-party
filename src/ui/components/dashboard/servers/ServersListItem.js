import React from 'react';
import PropTypes from 'prop-types';

// Aphrodite
import { css, StyleSheet } from 'aphrodite';

export default class ServersListItem extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string,
        distance: PropTypes.number
    };

    render() {
        const {
            name,
            distance,
        } = this.props;
        return (
            <div className={css(styles.container)}>
                <div className={css(styles.textContainer, styles.textLeft)}>
                    <h3 className={css(styles.text)}>
                        {name}
                    </h3>
                </div>
                <div className={css(styles.textContainer, styles.textRight)}>
                    <h3 className={css(styles.text)}>
                        {`${distance} km`}
                    </h3>
                </div>
            </div>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: '58px',
        display: 'flex',
        borderBottom: '1px solid #D8D8D8',
        boxSizing: 'border-box',
        ':hover': {
            backgroundColor: '#fcfcfc',
            color: '#fff',
        }
    },
    textContainer: {
        flex: '1 1 0%',
        display: 'flex',
        alignItems: 'center',
    },
    textLeft: {
        marginLeft: '14px',
        justifyContent: 'flex-start',
    },
    textRight: {
        marginRight: '17px',
        justifyContent: 'flex-end',
    },
    text: {
        color: '#999999',
        fontFamily: 'Roboto, sans-serif',
        fontSize: '14px',
        fontWeight: 300,
        lineHeight: '30px',
        letterSpacing: '1.4px',
    }
});
