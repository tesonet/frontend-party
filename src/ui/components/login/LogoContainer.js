import React from 'react';

// Aphrodite
import { css, StyleSheet } from 'aphrodite';

// Assets
import logoImage from '/assets/images/logo-testio.png';

export default class LogoContainer extends React.PureComponent {
    render() {

        return (
            <div className={css(styles.container)}>
                <img
                    src={logoImage}
                    alt="Testio."
                    className={css(styles.image)}
                />
            </div>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        maxWidth: '363px',
        height: '133px',
        textAlign: 'center',
    },
    image: {
        width: '100%',
        maxWidth: '246px',
        height: 'auto',
        '@media (max-width: 400px)': {
            width: '90%'
        },
    },
});
