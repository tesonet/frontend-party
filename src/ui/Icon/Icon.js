import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import {space} from 'styled-system'

import selection from './selection.json'

const defaultIconMap = selection.icons.reduce((acc, icon) => {
  acc[icon.properties.name] = icon

  return acc
}, {})

const validIconNames = Object.keys(defaultIconMap)

const Icon = ({color, size, name, iconMap, className}) => {
  const currentIcon = iconMap[name]

  return (
    <svg
      className={className}
      width={size}
      height={size}
      viewBox={`0 0 ${currentIcon.icon.width || '1024'} 1024`}
      xmlns="http://www.w3.org/2000/svg">
      {currentIcon.icon.paths.map((path, index) => {
        // eslint-disable-next-line
        return <path fill={color} key={index} d={path} />
      })}
    </svg>
  )
}

Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.oneOf(validIconNames).isRequired,
  size: PropTypes.number,
  // eslint-disable-next-line
  iconMap: PropTypes.object,
  color: PropTypes.string,
}

Icon.defaultProps = {
  className: null,
  iconMap: defaultIconMap,
  size: 16,
  color: null,
}

Icon.validIconNames = validIconNames

export default styled(Icon)`
  display: inline-block;
  vertical-align: middle;
  ${space}
`
