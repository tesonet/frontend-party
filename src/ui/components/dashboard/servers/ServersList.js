import React from 'react';
import PropTypes from 'prop-types';

// Aphrodite
import { css, StyleSheet } from 'aphrodite';

// Components
import ServersListItem from '/ui/components/dashboard/servers/ServersListItem';
import ServersListHeader from '/ui/components/dashboard/servers/ServersListHeader';
import Spin from '/ui/components/common/spin/';

export default class ServersList extends React.Component {
    static propTypes = {
        servers: PropTypes.arrayOf(PropTypes.shape({
            name: PropTypes.string,
            distance: PropTypes.number
        })),
        activeSort: PropTypes.string.isRequired,
        changeServersSort: PropTypes.func.isRequired,
    };

    render() {
        const { servers, activeSort, changeServersSort } = this.props;
        return (
            <div className={css(styles.container)}>
                <ServersListHeader
                    {...{
                        activeSort,
                        changeServersSort,
                    }}
                />
                {servers && servers.length ? (
                    <>
                        {servers.map((server, index) => (
                            <ServersListItem {...server} key={`${server.name}${index}`}/>
                        ))}
                    </>
                ) : (
                    <div className={css(styles.loadingContainer)}>
                        <Spin size={'large'} color={'grey'}/>
                    </div>
                )}
            </div>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: '100%',
    },
    loadingContainer: {
        width: '100%',
        marginTop: '50px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    }
});
