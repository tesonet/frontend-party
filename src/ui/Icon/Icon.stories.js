import React from 'react'
import {storiesOf} from '@storybook/react'
import {withKnobs, color} from '@storybook/addon-knobs'
import Box from '../Box'
import Icon from '.'

storiesOf('Icon', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <Box p={5}>{story()}</Box>)
  .add('with color picker', () => {
    return (
      <>
        <Box display="flex" alignItems="center" mb={2}>
          {Icon.validIconNames.map(name => {
            return (
              <Icon
                key={name}
                name={name}
                size={40}
                color={color('fill', '#000', 'GROUP-ID1')}
              />
            )
          })}
        </Box>
      </>
    )
  })
