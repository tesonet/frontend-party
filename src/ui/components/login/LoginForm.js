import React from 'react';
import PropTypes from 'prop-types';

// Aphrodite
import { css, StyleSheet } from 'aphrodite';

// Utils
import { validateLoginInputParams } from '/lib/utils';

// Components
import Icon from '/ui/components/common/icon';
import Spin from '/ui/components/common/spin';
import message from '/ui/components/common/message';

export default class LoginForm extends React.PureComponent {
    static propTypes = {
        loading: PropTypes.bool.isRequired,
        onLogin: PropTypes.func.isRequired,
    };

    state = {
        validationErrors: {},
    };

    render() {
        const { loading } = this.props;
        const { validationErrors } = this.state;
        return (
            <form
                onSubmit={this.handleLogin}
                className={css(styles.form)}
            >
                <div className={css(styles.inputWrapper)}>
                    <div className={css(styles.iconWrapper)}>
                        <Icon name={validationErrors['username'] ? 'attention' : 'user'}/>
                    </div>
                    <input
                        ref={this.username}
                        className={css(styles.input, styles.formText, validationErrors['username'] && styles.inputError)}
                        type='text'
                        name='username'
                        placeholder='Username'
                        onFocus={this.handleOnFocus}
                    />
                </div>
                <div className={css(styles.inputWrapper)}>
                    <div className={css(styles.iconWrapper)}>
                        <Icon name={validationErrors['password'] ? 'attention' : 'password'}/>
                    </div>
                    <input
                        ref={this.password}
                        className={css(styles.input, styles.formText, validationErrors['password'] && styles.inputError)}
                        type='password'
                        name='password'
                        placeholder='Password'
                        onFocus={this.handleOnFocus}
                    />
                </div>
                <div className={css(styles.inputWrapper)}>
                    <button
                        className={css(styles.submitButton, styles.formText)}
                        type="submit"
                        name="submit"
                    >
                        Log In
                        {loading && (
                            <Spin/>
                        )}
                    </button>
                </div>
            </form>
        );
    }

    username = React.createRef();
    password = React.createRef();

    // Remove Error highlights on focus
    handleOnFocus = (e) => {
        const validationErrors = { ...this.state.validationErrors };
        if (validationErrors[e.target.name]) {
            validationErrors[e.target.name] = null;
            this.setState({ validationErrors });
        }
    };

    // Handle user login
    handleLogin = (e) => {
        e.preventDefault();
        e.stopPropagation();

        const { onLogin } = this.props;
        const username = this.username.current.value;
        const password = this.password.current.value;

        const validationErrors = validateLoginInputParams({ username, password });

        if (validationErrors) {
            this.setState({ validationErrors }, () => {
                // Show validation errors to user
                // I'm using this approach because background is very dark and it's hard to see if I will add message under input fields
                Object.values(validationErrors).forEach((error) => {
                    message.attention(error.message);
                });
            });
        } else {
            onLogin({ username, password });
        }
    };
}

const styles = StyleSheet.create({
    inputWrapper: {
        position: 'relative',
        marginBottom: '18px',
    },
    formText: {
        fontFamily: 'Roboto, sans-serif',
        fontSize: '16px',
        letterSpacing: '0.4px',
    },
    input: {
        width: '100%',
        height: '56px',
        padding: '0 15px 0 55px',
        border: '1px solid #979797',
        borderRadius: '5px',
        boxShadow: '0 2px 4px 0 rgba(0,0,0,0.50)',
        boxSizing: 'border-box',
        fontWeight: 300,
        color: '#B3B3B3',
        ':active': {
            color: '#999',
        },
        ':focus': {
            outline: 0,
        }
    },
    inputError: {
        border: '1px solid #f64f64',
    },
    submitButton: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
        height: '58px',
        backgroundColor: ' #9FD533',
        border: 'none',
        borderRadius: '5px',
        fontWeight: 700,
        color: '#ffffff',
        ':hover': {
            backgroundColor: '#86b300',
            cursor: 'pointer',
        },
        ':focus': {
            outline: 0,
        },
        ':active': {
            border: 'none',
            backgroundColor: '#699600',
        },
    },
    iconWrapper: {
        position: 'absolute',
        top: '17px',
        left: '25px',
    },
});
