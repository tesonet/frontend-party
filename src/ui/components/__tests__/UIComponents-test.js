import React from 'react';

import { shallow } from 'enzyme';

// Components
import Spin from '../common/spin';
import Icon from '../common/icon';
import LoginForm from '../login/LoginForm';
import LogoContainer from '../login/LogoContainer';
import ServersList from '../dashboard/servers/ServersList';
import ServersListItem from '../dashboard/servers/ServersListItem';
import ServersListHeader from '../dashboard/servers/ServersListHeader';


describe('UI Components', () => {

    describe('Spin component', () => {
        const spinComponent = shallow(<Spin/>);
        test('renders correctly', () => {
            expect(spinComponent).toMatchSnapshot();
        });
    });

    describe('Icon component', () => {
        const iconComponent = shallow(<Icon name={'attention'}/>);
        test('renders correctly', () => {
            expect(iconComponent).toMatchSnapshot();
        });
    });

    describe('Login form component', () => {
        const loginFormComponent = shallow(<LoginForm loading={false} onLogin={() => {}} />);
        test('renders correctly', () => {
            expect(loginFormComponent).toMatchSnapshot();
        });
    });

    describe('Logo container component', () => {
        const logoContainerComponent = shallow(<LogoContainer />);
        test('renders correctly', () => {
            expect(logoContainerComponent).toMatchSnapshot();
        });
    });

    describe('ServersList component', () => {
        const serverMock = [
            {
                "name": "Germany #81",
                "distance": 650
            },
            {
                "name": "Lithuania #70",
                "distance": 210
            },
            {
                "name": "United States #98",
                "distance": 498
            },
        ];
        const activeSort = 'name';

        const changeServersSort = () => {}
        const serversListComponent = shallow(<ServersList {...{ servers: serverMock, activeSort, changeServersSort}}/>);
        test('renders correctly', () => {
            expect(serversListComponent).toMatchSnapshot();
        });
    });

    describe('Servers List Item component', () => {
        const serversListItemComponent = shallow(<ServersListItem />);
        test('renders correctly', () => {
            expect(serversListItemComponent).toMatchSnapshot();
        });
    });

    describe('ServersListHeader component', () => {
        const activeSort = 'name';
        const changeServersSort = () => {};
        const serversListHeader = shallow(<ServersListHeader {...{activeSort, changeServersSort}} />);
        test('renders correctly', () => {
            expect(serversListHeader).toMatchSnapshot();
        });
    });

});
