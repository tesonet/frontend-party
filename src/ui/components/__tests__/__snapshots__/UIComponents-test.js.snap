// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Components Icon component renders correctly 1`] = `
<div
  className=""
  style={
    Object {
      "height": "19px",
      "width": "17px",
    }
  }
>
  <img
    alt="attention"
    className="image_190hxbq"
    src="test-file-stub"
  />
</div>
`;

exports[`UI Components Login form component renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="inputWrapper_ebxu7m"
  >
    <div
      className="iconWrapper_17iirxi"
    >
      <Index
        name="user"
        style={
          Object {
            "height": "19px",
            "width": "17px",
          }
        }
      />
    </div>
    <input
      className="input_1wh6aqi-o_O-formText_ggyguu"
      name="username"
      onFocus={[Function]}
      placeholder="Username"
      type="text"
    />
  </div>
  <div
    className="inputWrapper_ebxu7m"
  >
    <div
      className="iconWrapper_17iirxi"
    >
      <Index
        name="password"
        style={
          Object {
            "height": "19px",
            "width": "17px",
          }
        }
      />
    </div>
    <input
      className="input_1wh6aqi-o_O-formText_ggyguu"
      name="password"
      onFocus={[Function]}
      placeholder="Password"
      type="password"
    />
  </div>
  <div
    className="inputWrapper_ebxu7m"
  >
    <button
      className="submitButton_15v5db7-o_O-formText_ggyguu"
      name="submit"
      type="submit"
    >
      Log In
    </button>
  </div>
</form>
`;

exports[`UI Components Logo container component renders correctly 1`] = `
<div
  className="container_1vg9bv0"
>
  <img
    alt="Testio."
    className="image_1pknxbh"
    src="test-file-stub"
  />
</div>
`;

exports[`UI Components Servers List Item component renders correctly 1`] = `
<div
  className="container_izy9jc"
>
  <div
    className="textContainer_ovz5au-o_O-textLeft_4fgp8a"
  >
    <h3
      className="text_1s0zdaf"
    />
  </div>
  <div
    className="textContainer_ovz5au-o_O-textRight_deghhp"
  >
    <h3
      className="text_1s0zdaf"
    >
      undefined km
    </h3>
  </div>
</div>
`;

exports[`UI Components ServersList component renders correctly 1`] = `
<div
  className="container_190hxbq"
>
  <ServersListHeader
    activeSort="name"
    changeServersSort={[Function]}
  />
  <ServersListItem
    distance={650}
    key="Germany #810"
    name="Germany #81"
  />
  <ServersListItem
    distance={210}
    key="Lithuania #701"
    name="Lithuania #70"
  />
  <ServersListItem
    distance={498}
    key="United States #982"
    name="United States #98"
  />
</div>
`;

exports[`UI Components ServersListHeader component renders correctly 1`] = `
<div
  className="container_1my5rsq"
>
  <div
    className="headerTitleContainer_ovz5au-o_O-headerTitleLeft_4fgp8a"
  >
    <h3
      className="headerText_1twnfzb-o_O-active_u5j6yl"
      onClick={[Function]}
    >
      SERVER
    </h3>
  </div>
  <div
    className="headerTitleContainer_ovz5au-o_O-headerTitleRight_deghhp"
  >
    <h3
      className="headerText_1twnfzb"
      onClick={[Function]}
    >
      DISTANCE
    </h3>
  </div>
</div>
`;

exports[`UI Components Spin component renders correctly 1`] = `
<div
  className="lds-dual-ring small white"
/>
`;
